### DISCUSSION TITLE
Upgrade to support pydantic 2.0+

### DISCUSSION THREAD

**Discussion by jessereich-mcs** [tiangolo/pydantic-sqlalchemy#100] [Questions] [answered]
Currently pinned to pydantic<2.0.0 but pydantic 2.0 has been out for a few months.

**Comment by parikls** [comment]
+1 for this

**Comment by ss1917** [comment]
+1 for this

**Comment by yanghongfei** [comment]
+1 for this

**Comment by ops-coffee** [comment]
+1 for this

**Comment by cyancow** [comment]
+1 for this

**Comment by wezhai** [comment]
If you are using pydantic v2 and sqlalchemy 2.0 and above, try this [sqlalchemy-to-pydantic](https://pypi.org/project/sqlalchemy-to-pydantic/)

**Comment by jamesbraza** [comment]
For those looking for a link to the source code behind [sqlalchemy-to-pydantic](https://pypi.org/project/sqlalchemy-to-pydantic/), it's here: https://github.com/wezhai/sqlalchemy-to-pydantic. Linking sqlalchemy-to-pydantic's source to PyPI was requested in https://github.com/wezhai/sqlalchemy-to-pydantic/issues/1

**Comment by realazizk** [comment]
@wezhai that requires sqlalchemy 2 🤔

**Comment by realazizk** [comment]
this make from_orm work again https://github.com/mohamed-aziz/pydantic-sqlalchemy/commit/4b1821c418014e26c53449a3800beba318c130f3

**Comment by tiangolo** [comment, marked as answer]
Thanks for the patience with my reply! :sweat_smile:

This would be a feature request.

I deprecated this library a while ago in favor of SQLModel which would solve all the same use cases in a much better way, and I'm directing all the effort in that direction.

SQLModel uses Pydantic models to define both data and table (SQL) models, so it's easier to do all these things. :rocket: 

Also, this library only supports SQLAlchemy 1.4.x, not 2.x. :melting_face:
